nemo_curator/modules/config.py

```diff
@@ -84,9 +84,15 @@ def __post_init__(self):
             raise ValueError(
                 "Finding fuzzy duplicates requires a cache directory accessible via all workers to store intermediates"
             )
-        if not self.false_positive_check:
-            raise NotImplementedError(
-                "Skipping false positive checks is not supported at the moment"
+        if self.false_positive_check:
+            warnings.warn(
+                "Identifying false positives during the Minhash deduplication is computationally expensive."
+                " For improved performance consider setting this to False"
+            )
+        if not self.false_positive_check and self.char_ngrams < 20:
+            warnings.warn(
+                "Using a small char_ngrams value might lead to a large number (~5%) of false positives during deduplication."
```

Comment:
Theoretical FP rate with respect to Jaccard similarity can be computed like this, given the parameters of minhash LSH: https://github.com/ekzhu/datasketch/blob/5512549871d29c55b3cd8e99a79fcbd14859b77d/datasketch/lsh.py#L12

The issue with Jaccard similarity is that it itself is just a proxy for document similarity, quality of which depends on the chosen size and type of n-grams. Are there any studies that would compare Jaccard similarity with, say, edit similarity? Or is there even a theoretical way to do that?

We did some comparisons of minhash LSHs deduplication with word-based 5-grams vs 13-grams, and we found that FP rates with respect to Jaccard similarity was comparable with a theoretical value, but edit similarity FP rate was much smaller with 13-grams.

Reply:
Agreed. From my understanding the formula helps estimate FP's based on the number of hashes and buckets, but as you mentioned doesn't take into account ngram sizes since there is no standard on what is the baseline ngram size for overlap (w.r.t jaccard) or equivalent edit distance that makes 2 document duplicates.

>Are there any studies that would compare Jaccard similarity with, say, edit similarity? Or is there even a theoretical way to do that?

I need to check but am not immediately aware of any studies/approach.

For the time being the 5% number quoted here is empirical evidence from some of our internal experiments at smaller ngram sizes (1 word/5char ngrams) where skipping the false positive check leads to an additional 5% documents being marked as duplicate. The % goes down when using larger ngrams (20 chars or above).